zulipterminal/ui_tools/views.py

```diff
@@ -1492,6 +1493,58 @@ def keypress(self, size: urwid_Size, key: str) -> str:
         return super().keypress(size, key)
 
 
+class TopicInfoView(PopUpView):
+    def __init__(self, controller: Any, stream_id: int, topic: str) -> None:
+        self.stream_id = stream_id
+        self.controller = controller
+        stream = controller.model.stream_dict[stream_id]
+        self.topic_name = topic
+        stream_name = stream["name"]
+        self.resolve_topic_btn_lbl = ""
+
+        title = f"{stream_name} {STREAM_TOPIC_SEPARATOR} {self.topic_name}"
+
+        topic_info_content = []  # type: PopUpViewTableContent
+
+        popup_width, column_widths = self.calculate_table_widths(
+            topic_info_content, len(title)
+        )
+
+        if self.topic_name.startswith(RESOLVED_TOPIC_PREFIX):
+            self.resolve_topic_setting_btn_lbl = "Unresolve Topic"
+        else:
+            self.resolve_topic_setting_btn_lbl = "Resolve Topic"
+        resolve_topic_setting = urwid.Button(
+            self.resolve_topic_setting_btn_lbl,
+            self.toggle_resolve_status,
+        )
+        resolve_topic_setting._w = urwid.AttrMap(
+            urwid.SelectableIcon(self.resolve_topic_setting_btn_lbl), None, "selected"
+        )
+        popup_width = max(
+            popup_width,
+            len(resolve_topic_setting.label) + 4,
+        )
+
+        self.widgets = self.make_table_with_categories(
+            topic_info_content, column_widths
+        )
+
+        self.widgets.append(resolve_topic_setting)
+        super().__init__(controller, self.widgets, "TOPIC_DESC", popup_width, title)
+
+    def toggle_resolve_status(self, args: Any) -> None:
+        self.resolve_topic_btn_lbl = "Button clicked"
```

Comment:
Probably exiting out of the popup (after executing current functionality) as soon as you press the button would help with this.

Reply:
So, like if we press the button, the popup will exit and the topic will be (un)resolved.
But, we will be also adding different settings to the topic popup in the future. So if a user wants to (un)resolve topic and change any other setting, user again has to toggle the popup?

Reply:
Since we'll place 1 time use features here like `Mute Topic`, `Mark topic as read`, `Copy Topic Link`, `Resolve Topic`, it make sense to close it when you activate the button.

Reply:
We can revisit this, but a button like Rohitth indicates may be simplest for now.

Note that for the stream info, we have a check-box which updates, but as I think I mentioned before, that's a user setting so isn't as big an issues if eg. updated repeatedly via a check-box.